pkg/reconciler/app/reconciler.go

```diff
@@ -186,6 +187,33 @@ func (r *Reconciler) ApplyChanges(ctx context.Context, app *v1alpha1.App) error
 		app.Status.PropagateKnativeServiceStatus(actual)
 	}
 
+	// Route Reconciler
+	{
+		r.Logger.Info("reconciling Routes")
+		condition := app.Status.KnativeServiceCondition()
+		desiredRoutes, err := resources.MakeRoutes(app, space)
+		if err != nil {
+			return condition.MarkTemplateError(err)
+		}
+
+		for _, desired := range desiredRoutes {
+			actual, err := r.routeLister.Routes(desired.GetNamespace()).Get(desired.Name)
+			if apierrs.IsNotFound(err) {
+				// Route doesn't exist, make one.
+				actual, err = r.KfClientSet.KfV1alpha1().Routes(desired.GetNamespace()).Create(&desired)
+				if err != nil {
+					return condition.MarkReconciliationError("creating", err)
+				}
+			} else if err != nil {
+				return condition.MarkReconciliationError("getting latest", err)
+			} else if actual, err = r.reconcileRoute(&desired, actual); err != nil {
+				return condition.MarkReconciliationError("updating existing", err)
+			}
+
+			// TODO: app should propogate v1alpha1.Route status
```

Comment:
Without a dummy one at the end it's impossible to get back to a good state once failed.